x-pack/plugins/security_solution/public/timelines/components/timeline/tabs/shared/use_timeline_columns.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEmpty } from 'lodash/fp';
+import { useCallback, useMemo } from 'react';
```

Comment:
Thanks for this abstraction 🥳 .

Reply:
Yea, no worries. But thought about it last night and I should just manage all of this via redux and selectors instead of using this hook. May do a more focused move of custom hooks => useSelector in a tech debt PR I think